src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/SdkRuntimePacksTests.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.SourceBuild.SmokeTests
+{
+    public class SdkRuntimePacksTests : SmokeTests
+    {
+        private static readonly string _nuGetCacheFolder = "nuget-cache";
+
+        // `dotnet new webapp` will have dependency on the two runtime packs that are distributed with the SDK:
+        //   - Microsoft.AspNetCore.App.Runtime.<rid>
+        //   - Microsoft.NETCore.App.Runtime.<rid>
+        private static readonly string _templateName = DotNetTemplate.WebApp.GetName();
+
+        public SdkRuntimePacksTests(ITestOutputHelper outputHelper) : base(outputHelper) { }
+
+        [Fact]
+        public void VerifyRuntimePacksForSelfContained()
```

Comment:
Technically existing scenario tests already do a self-contained publish, including a publish with a specific RID or ReadyToRun - https://github.com/dotnet/installer/blob/1d7b99dc1f26b093b2cf8529597e4b4039575cb7/src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/TestScenario.cs#L51

I just needed to do some specific validation afterwards which I did not see in the TestScenario and it was relatively cheap to achieve this way. Not sure there would be value in adding this validation behind a flag

Reply:
From my point of view `TestScenario` achieves a different goal - it executes a specific path where the path succeeding is the test. We are making sure that a tested command can be executed.

This test is verifying a specific aspect for which the command is just a setup, not the test itself. It differs in concept & execution and doesn't feel to me like it would fit in the `TestScenario`.

Multiple configuration inputs can be solved quite easily through XUnit attributes. Scenarios covered could also be displayed / summarized if needed outside of just having them in one file

Reply:
Discussed offline - agreed to move the test under `TestScenario` and expand `DotNetHelper` utility to accommodate the current use-case